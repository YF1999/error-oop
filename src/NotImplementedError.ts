import { IErrorArguments, IErrorOptions, NativeError } from './Native';
import { InternalErrorTool } from './InternalTools';

export interface INotImplementedErrorOptions extends IErrorOptions {}

/**
 * Applicable when a requested method or operation is not implemented.
 */
export class NotImplementedError extends NativeError {
    public constructor();
    /**
     * @param message The error message that explains the reason for this error.
     */
    public constructor(message: string);
    /**
     * @param message The error message that explains the reason for this error.
     * @param innerError The error that is the cause of the current error. Stack trace will be appended.
     */
    public constructor(message: string, innerError: Error);
    /**
     * @param options The constructor options.
     */
    public constructor(options: INotImplementedErrorOptions);

    public constructor(...args: IErrorArguments<INotImplementedErrorOptions>) {
        super(InternalErrorTool.parseErrorArguments([], ...args));
    }
}
