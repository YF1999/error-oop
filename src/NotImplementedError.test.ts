import { NotImplementedError } from './NotImplementedError';

standardTestError(NotImplementedError, 'NotImplementedError', 'error');
