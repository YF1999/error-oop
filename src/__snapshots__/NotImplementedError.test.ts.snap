// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test \`NotImplementedError\` own keys 1`] = `
Array [
  "stack",
]
`;

exports[`Test \`NotImplementedError\` own keys 2`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`NotImplementedError\` own keys 3`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`NotImplementedError\` own keys 4`] = `
Array [
  "stack",
  "message",
]
`;
